scripts/tasks_retrival/gene_disease_association_task_creation.py

```diff
@@ -0,0 +1,275 @@
+            gda_df = pd.read_parquet(input_file + link_ars)
+            res.append(gda_df)
+            part_ind = part_ind + 1
+        except Exception as exception:
+            if not len(res):
+                raise RuntimeError(f"could not read from {input_file}") from exception
+            file_exist = False
+            break
+
+    return pd.concat(res)
+
+
+def extract_outcome_df(
+    downloaded_dataframe: pd.DataFrame, target_column_name: str = COLUMN_OF_OUTCOME
+) -> pd.DataFrame:
+    """Splice the two columns of interest into a new data frame, rename and extract HLC Class from the name."""
+    outcomes = pd.DataFrame({"Outcomes": downloaded_dataframe[target_column_name]})
+    # make sure there are no extra spaces around the values and return
+    return outcomes.map(lambda x: x.strip() if isinstance(x, str) else x)
+
+
+def report_task(df, task_dir_name, max_counts=10):
+    print(f"Task saved to {task_dir_name}/\n")
+    print(f"total samples = {len(df)}")
+
+
+@click.command("cli", context_settings={"show_default": True})
+@click.option(
+    "--task-name",
+    "-n",
+    type=click.STRING,
+    help="name for the specific task",
+    default=TASK_NAME,
+)
+@click.option(
+    "--input-file",
+    "-i",
+    type=click.STRING,
+    help="path local data file.",
+    default=None,
+)
+@click.option(
+    "--allow-downloads",
+    help=f"download files directly from {DATA_URL}",
+    type=click.BOOL,
+    default=False,
+)
+@click.option(
+    "--main-task-directory",
+    "-m",
+    type=click.STRING,
+    help="The task root directory.  Will not be created.",
+    default="./tasks",
+)
+@click.option(
+    "--association-type",
+    type=click.STRING,
+    help="The type of association to save",
+    default="genetic_association",
+)
+@click.option(
+    "--verbose",
+    "-v",
+    is_flag=True,
+)
+def main(
+    task_name,
+    main_task_directory,
+    association_type,
+    input_file,
+    allow_downloads,
+    verbose,
+):
+    input_path_or_url = verify_source_of_data(
+        input_file, allow_downloads=allow_downloads
+    )
+
+    if verbose:
+        print(f"creating the {task_name} task")
+        print("This may take several minutes")
+    downloaded_dataframe = get_gene_drug_assosiation_data(input_file=input_path_or_url)
+    gene_drug_assosiation_df = downloaded_dataframe.loc[
+        downloaded_dataframe["datatypeId"] == association_type, :
+    ]
+    symbols = get_symbols(gene_drug_assosiation_df[COLUMN_OF_SYMBOLS])
+    diseaseId = gene_drug_assosiation_df["diseaseId"]
+    entities = pd.concat(
+        [symbols.reset_index(drop=True), diseaseId.reset_index(drop=True)], axis=1
+    )
+    outcomes = extract_outcome_df(gene_drug_assosiation_df)
+
+    assert len(entities) == len(
+        outcomes
+    ), " the lengths of the concatinated files does not match"
+
+    # entered by the user.
```

Comment:
I think the following should also be in a single method: `dump_task(entities, outcomes, path_name)`.  form line to 258 to line 268.